src/main/java/org/truffleruby/core/string/InterpolatedStringNode.java

```diff
@@ -28,10 +28,13 @@
 
     private final Rope emptyRope;
 
-    public InterpolatedStringNode(ToSNode[] children, Encoding encoding) {
+    private boolean frozen;
```

Comment:
Ah sorry - this _must_ be final, so that line 53, below, can be removed as dead-code if the string is not frozen in the common case.